### PR TITLE
Store Services: Add 'Cancel' button to add credit card modal, and remove header

### PR DESCRIPTION
**Pull request by dechov**
Fixes https://github.com/Automattic/wp-calypso/issues/24459

Removes "Back"  button in the Add Credit Card modal's header, and replaces it with a "Cancel" button in the footer:

add-credit-card-form

`<HeaderCake>` couldn't be rendered without a "← Back" button, without modifying that component. I chose to omit it and lose the title, which I think is OK — now the `<CreditCardForm>` block is rendered alone in the modal — but if not, what is a good alternative?

### CODE REVIEW

**Comment by kellychoffman** [comment]
> but if not, what is a good alternative?

It would be ideal if we could render a modal with a header that didn't require a back button. Is that possible? Perhaps the [Formatted header](https://wpcalypso.wordpress.com/devdocs/design/formatted-header)? I couldn't find a modal/dialog pattern in the devdocs 🤔

**Comment by dechov** [comment]
Trying `<FormattedHeader>` | Original | Margin evened out
-- | -- | --
`headerText` | with-formatted-header | screen shot 2018-05-08 at 9 23 43 am (LGTM)
With `subHeaderText` (text itself would need editing) | with-subheader-too | screen shot 2018-05-08 at 9 09 44 am

@kellychoffman 👍❓

**Comment by dechov** [comment]
Glad you asked 😄— added another optional prop `heading` to the credit card form block.

screen shot 2018-05-08 at 5 49 37 pm

Any tweaks to styling/capitalization? Title case, for comparison, looks like:

screen shot 2018-05-08 at 5 54 03 pm

**Comment by kellychoffman** [comment]
Ding ding ding! 

Non title case is a go. :shipit:

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js [hunk 1]
```diff
@@ -48,13 +47,14 @@ class AddCardDialog extends Component {
...
 				onClose={ onClose }
 			>
-				<HeaderCake onClick={ onClose }>{ translate( 'Add Credit Card' ) }</HeaderCake>
 				<CreditCardForm
 					createCardToken={ this.createCardToken }
 					recordFormSubmitEvent={ this.recordFormSubmitEvent }
 					saveStoredCard={ this.props.addStoredCard }
 					successCallback={ onClose }
 					showUsedForExistingPurchasesInfo={ true }
+					heading={ translate( 'Add credit card' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: Hi! I've found a possible matching string that has already been translated **41** times:
`translate( 'Add Credit Card' )` *ES Score: 14*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/24459-store-shipping-settings-add-credit-card-modal-close-button)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (DanReyLop) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: I guess you should use `Title Case` then, for consistency?

**Review by DanReyLop — APPROVED** [review]
Looks good, probably you should switch to `Title Case`, see my comment. Pre-approving :shipit:

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js (hunk 1 above)
Comment (dechov) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: 👍 Changed to title case in code but styled as non title case for display in https://github.com/Automattic/wp-calypso/pull/24715/commits/a6463238d11d09319f327adb3b368346138a6434

Comment (DanReyLop) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: That's not worth it IMHO, we should just use title case instead of CSS trickery. Calypso is using title case in other places, that's why the exact string `Add Credit Card` is used in many more places.

Comment (dechov) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: I would think it's less trickery and more just styling, and I too preferred the sentence case (though I'm not sure why), but omitting those rules would be simpler and I don't mind the title case so much. @kellychoffman are you OK with the title case – and if so, can we leave it as is in the screenshot above or adjust the appearance to look more like a title?

Comment (v18) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: Hi everyone, this discussion is a bit problematic to me.

Already-done translations should not dictate our UI. The bot is here to help us to make use of existing translations when available, but it's not necessary to always follow the suggestions made (I've been told this by the i18n team). Case-in-point, it even says "even if not implemented" right in the bot response.

@kellychoffman already decided and pointed out that we need to use non-title case for this UI. We should not change it based on a translation suggestion.

I don't think we should use CSS to change the title case since the localization for different languages may be different (this is not 100% verified, but a general rule of thumb when considering i18n).

We need to use the new non-title case string.

TLDR: there was already a design decision made, we should not change is based on the suggestion of a translation bot.

Comment (dechov) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/add-credit-card-modal.js]: OK, totally on board with stylistic design _decision_ superseding i18n _suggestion_ by robot 😄

> the localization for different languages may be different 

Hadn't thought this through well enough before making that change. In German, for instance, the lowercase noun might just look wrong. You and @DanReyLop are right – this was an inappropriate use of CSS – now reverted/reset.
